### ISSUE TITLE
在whisper官方预训练模型上进行aishell测试集推理，cer>100%

### ISSUE THREAD

**Issue by JiangSYKY** [k2-fsa/icefall#2072]
作者您好，非常感谢你们的开源贡献。我在复现推理以下结果时出现了问题
fine-tuning results on Aishell test set on whisper medium, large-v2, large-v3
test (before fine-tuning)	test (after fine-tuning)	comment
medium	7.23	3.27	--epoch 10 --avg 4, ddp
large-v2	6.56	2.47	--epoch 10 --avg 6, deepspeed zero stage1
large-v3	6.06	2.84	--epoch 5 --avg 3, deepspeed zero stage1

使用aishell test，基于large_V2，使用huggingface下载的：/root/autodl-tmp/icefall/egs/aishell/ASR/icefall_asr_aishell_whisper/exp_large_v2/epoch-10-avg-6.pt得到了test (after fine-tuning）的正确结果，但在test (before fine-tuning)中，cer>100%
使用的推理代码：
python3 ./whisper/decode.py \
  --exp-dir whisper/exp_large_v2_pre \
  --model-name large-v2 \
  --epoch -1 --avg 1 \
  --remove-whisper-encoder-input-length-restriction False \
  --beam-size 10 --max-duration 15
然后我的large-v2模型的路径是：/root/.cache/whisper/large-v2.pt
是通过pip install -r whisper/requirements.txt自动下载的。
请问是哪里出了问题呢？--epoch -1时调用的是什么模型呢？是我这里加载的预训练模型不正确吗？

**Comment by JiangSYKY** [comment]
具体的：
2026-03-02 10:06:54,450 WETEXT INFO found existing fst: /root/miniconda3/envs/icefall/lib/python3.10/site-packages/tn/zh_tn_tagger.fst
2026-03-02 10:06:54,450 INFO [processor.py:95] found existing fst: /root/miniconda3/envs/icefall/lib/python3.10/site-packages/tn/zh_tn_tagger.fst
2026-03-02 10:06:54,450 WETEXT INFO                     /root/miniconda3/envs/icefall/lib/python3.10/site-packages/tn/zh_tn_verbalizer.fst
2026-03-02 10:06:54,450 INFO [processor.py:96]                     /root/miniconda3/envs/icefall/lib/python3.10/site-packages/tn/zh_tn_verbalizer.fst
2026-03-02 10:06:54,450 WETEXT INFO skip building fst for zh_normalizer ...
2026-03-02 10:06:54,450 INFO [processor.py:97] skip building fst for zh_normalizer ...
2026-03-02 10:06:54,733 INFO [decode0.py:426] Decoding started
2026-03-02 10:06:54,733 INFO [decode0.py:433] device: cuda
/root/miniconda3/envs/icefall/lib/python3.10/site-packages/whisper/__init__.py:150: FutureWarning: You are using `torch.load` with `weights_only=False` (the current default value), which uses the default pickle module implicitly. It is possible to construct malicious pickle data which will execute arbitrary code during unpickling (See https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). In a future release, the default value for `weights_only` will be flipped to `True`. This limits the functions that could be executed during unpickling. Arbitrary objects will no longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the user via `torch.serialization.add_safe_globals`. We recommend you start setting `weights_only=True` for any use case where you don't have full control of the loaded file. Please open an issue on GitHub for any issues related to this experimental feature.
  checkpoint = torch.load(fp, map_location=device)
2026-03-02 10:07:19,326 INFO [decode0.py:481] Number of model parameters: 1541384960
2026-03-02 10:07:19,326 INFO [asr_datamodule.py:413] About to get test cuts
2026-03-02 10:07:19,328 INFO [asr_datamodule.py:374] About to create test dataset
2026-03-02 10:07:22,166 INFO [decode0.py:344] batch 0/?, cuts processed until now is 2
2026-03-02 10:08:36,287 INFO [decode0.py:344] batch 100/?, cuts processed until now is 258
2026-03-02 10:09:36,961 INFO [decode0.py:344] batch 200/?, cuts processed until now is 501
2026-03-02 10:10:43,260 INFO [decode0.py:344] batch 300/?, cuts processed until now is 726
2026-03-02 10:12:06,167 INFO [decode0.py:344] batch 400/?, cuts processed until now is 950
2026-03-02 10:13:11,048 INFO [decode0.py:344] batch 500/?, cuts processed until now is 1167
2026-03-02 10:14:18,346 INFO [decode0.py:344] batch 600/?, cuts processed until now is 1411
2026-03-02 10:15:18,767 INFO [decode0.py:344] batch 700/?, cuts processed until now is 1646
2026-03-02 10:16:28,684 INFO [decode0.py:344] batch 800/?, cuts processed until now is 1876
2026-03-02 10:17:28,327 INFO [decode0.py:344] batch 900/?, cuts processed until now is 2095
2026-03-02 10:18:27,483 INFO [decode0.py:344] batch 1000/?, cuts processed until now is 2333
2026-03-02 10:19:34,386 INFO [decode0.py:344] batch 1100/?, cuts processed until now is 2582
2026-03-02 10:20:38,593 INFO [decode0.py:344] batch 1200/?, cuts processed until now is 2805
2026-03-02 10:21:42,793 INFO [decode0.py:344] batch 1300/?, cuts processed until now is 3041
2026-03-02 10:22:56,461 INFO [decode0.py:344] batch 1400/?, cuts processed until now is 3284
2026-03-02 10:23:56,052 INFO [decode0.py:344] batch 1500/?, cuts processed until now is 3502
2026-03-02 10:25:04,096 INFO [decode0.py:344] batch 1600/?, cuts processed until now is 3731
2026-03-02 10:26:11,909 INFO [decode0.py:344] batch 1700/?, cuts processed until now is 3967
2026-03-02 10:27:12,091 INFO [decode0.py:344] batch 1800/?, cuts processed until now is 4204
2026-03-02 10:28:19,189 INFO [decode0.py:344] batch 1900/?, cuts processed until now is 4458
2026-03-02 10:29:28,289 INFO [decode0.py:344] batch 2000/?, cuts processed until now is 4687
2026-03-02 10:30:35,342 INFO [decode0.py:344] batch 2100/?, cuts processed until now is 4921
2026-03-02 10:31:40,381 INFO [decode0.py:344] batch 2200/?, cuts processed until now is 5158
2026-03-02 10:32:47,359 INFO [decode0.py:344] batch 2300/?, cuts processed until now is 5416
2026-03-02 10:33:44,010 INFO [decode0.py:344] batch 2400/?, cuts processed until now is 5650
2026-03-02 10:35:10,520 INFO [decode0.py:344] batch 2500/?, cuts processed until now is 5881
2026-03-02 10:36:05,633 INFO [decode0.py:344] batch 2600/?, cuts processed until now is 6098
2026-03-02 10:37:07,341 INFO [decode0.py:344] batch 2700/?, cuts processed until now is 6327
2026-03-02 10:38:07,633 INFO [decode0.py:344] batch 2800/?, cuts processed until now is 6564
2026-03-02 10:39:17,294 INFO [decode0.py:344] batch 2900/?, cuts processed until now is 6788
2026-03-02 10:40:31,712 INFO [decode0.py:344] batch 3000/?, cuts processed until now is 7016
2026-03-02 10:41:21,615 INFO [decode0.py:363] The transcripts are stored in whisper/exp_large_v2_pre/recogs-aishell-beam-search-epoch--1-avg-1.txt
2026-03-02 10:41:21,903 INFO [utils.py:714] [aishell-beam-search] %WER 165.91% [173813 / 104765, 69657 ins, 45796 del, 58360 sub ]
2026-03-02 10:41:22,377 INFO [decode0.py:385] Wrote detailed error stats to whisper/exp_large_v2_pre/errs-aishell-beam-search-epoch--1-avg-1.txt
2026-03-02 10:41:22,378 INFO [decode0.py:399] 
For aishell, CER of different settings are:
beam-search     165.91  best for aishell

**Comment by csukuangfj** [comment]
@yuekaizhang  Can you have a look?

**Comment by yuekaizhang** [comment]
@JiangSYKY  看你提供的命令目前没发现有什么问题。 After fine-tuning 的结果能对上，就可以了。

Before fine-tuning 的结果，实在对不上的话，可以参考一下别的论文里面解码出来的结果, 比如这个 https://github.com/yeyupiaoling/Whisper-Finetune； 或者可以考虑使用别的 whisper 推理库去跑一下。
